api-flask/app/zonal_stats.py

```diff
@@ -1,26 +1,77 @@
 """Calulate zonal statistics and return a json or a geojson."""
 import logging
 from collections import defaultdict
+from datetime import datetime, timedelta
 from json import dump, load
 
+from app.caching import get_geojson_file
 from app.timer import timed
 
 from rasterstats import zonal_stats
 
-from shapely.geometry import mapping, shape
+import requests
+
+from shapely.geometry import GeometryCollection, mapping, shape
 from shapely.ops import cascaded_union
 
+from werkzeug.exceptions import InternalServerError
+
 logger = logging.getLogger(__name__)
 
 
 DEFAULT_STATS = ['min', 'max', 'mean', 'median']
 
 
-def _extract_features_properties(zones):
-    with open(zones) as json_file:
-        geojson_data = load(json_file)
-    features = geojson_data.get('features', [])
-    return [feature['properties'] for feature in features]
+def get_wfs_response(wfs_params, zones):
+    """
+    Execute Web Feature Service (WFS) request to external OGC server.
+
+    This request returns geospatial features that match required filters within cql_filter param.
+
+    https://docs.geoserver.org/stable/en/user/services/wfs/reference.html
+    """
+    geoms = [shape(f['geometry']) for f in zones.get('features')]
+    envelope = GeometryCollection(geoms).envelope.wkt
+
+    # We make a list and then join
+    cql_filter = ['INTERSECTS(shape, {})'.format(envelope)]
+
+    if 'time' in wfs_params.keys():
+        from_date = datetime.strptime(wfs_params.get('time'), '%Y-%m-%d')
+        to_date = from_date + timedelta(days=1)
+
+        time_filters = ['pub_date AFTER {}'.format(from_date.isoformat()),
+                        'pub_date BEFORE {}'.format(to_date.isoformat())]
+
+        cql_filter.extend(time_filters)
+
+    cql_filter = ' AND '.join(cql_filter)
+
+    params = {
+        'service': 'WFS',
+        'version': '1.0.0',
+        'request': 'GetFeature',
+        'typeName': wfs_params.get('layer_name'),
+        'outputFormat': 'application/json',
+        'cql_filter': cql_filter
+    }
+
+    resp = requests.get(wfs_params.get('url'), params)
+    if resp.status_code != 200:
+        logger.error(resp.content)
+        err_message = 'Received status code from WFS request: {}'.format(resp.status_code)
+
+        raise InternalServerError(err_message)
+
+    # A WFS response should be always a json response.
```

Comment:
-> should always be

Reply:
Done. included in e7ba78168bbd18bacd50227d6d28ad7117fdf44a